test/reducedim.jl

```diff
@@ -117,22 +117,201 @@ R = reducedim((a,b) -> a+b, [1 2; 3 4], 2, 0.0)
 rt = Base.return_types(reducedim, Tuple{Function, Array{Float64, 3}, Int, Float64})
 @test length(rt) == 1 && rt[1] == Array{Float64, 3}
 
+@testset "empty cases" begin
+    A = Array{Int}(0,1)
+    @test sum(A) === 0
+    @test prod(A) === 1
+    @test_throws ArgumentError minimum(A)
+    @test_throws ArgumentError maximum(A)
+    @test var(A) === NaN
 
-## findmin/findmax
-A = [1.0 3.0 6.0;
+    @test isequal(sum(A, 1), zeros(Int, 1, 1))
+    @test isequal(sum(A, 2), zeros(Int, 0, 1))
+    @test isequal(sum(A, (1, 2)), zeros(Int, 1, 1))
+    @test isequal(sum(A, 3), zeros(Int, 0, 1))
+    @test isequal(prod(A, 1), ones(Int, 1, 1))
+    @test isequal(prod(A, 2), ones(Int, 0, 1))
+    @test isequal(prod(A, (1, 2)), ones(Int, 1, 1))
+    @test isequal(prod(A, 3), ones(Int, 0, 1))
+    @test isequal(var(A, 1), fill(NaN, 1, 1))
+    @test isequal(var(A, 2), fill(NaN, 0, 1))
+    @test isequal(var(A, (1, 2)), fill(NaN, 1, 1))
+    @test isequal(var(A, 3), fill(NaN, 0, 1))
+
+    for f in (minimum, maximum)
+        @test_throws ArgumentError f(A, 1)
+        @test isequal(f(A, 2), zeros(Int, 0, 1))
+        @test_throws ArgumentError f(A, (1, 2))
+        @test isequal(f(A, 3), zeros(Int, 0, 1))
+    end
+    for f in (findmin, findmax)
+        @test_throws ArgumentError f(A, 1)
+        @test isequal(f(A, 2), (zeros(Int, 0, 1), zeros(Int, 0, 1)))
+        @test_throws ArgumentError f(A, (1, 2))
+        @test isequal(f(A, 3), (zeros(Int, 0, 1), zeros(Int, 0, 1)))
+    end
+
+end
+## findmin/findmax/minumum/maximum
+
+A = [1.0 5.0 6.0;
      5.0 2.0 4.0]
 for (tup, rval, rind) in [((1,), [1.0 2.0 4.0], [1 4 6]),
                           ((2,), reshape([1.0,2.0], 2, 1), reshape([1,4], 2, 1)),
                           ((1,2), fill(1.0,1,1),fill(1,1,1))]
     @test findmin(A, tup) == (rval, rind)
     @test findmin!(similar(rval), similar(rind), A) == (rval, rind)
+    @test string(minimum(A, tup)) == string(rval)
+    @test string(minimum!(similar(rval), A)) == string(rval)
+    @test string(minimum!(copy(rval), A, init=false)) == string(rval)
 end
 
-for (tup, rval, rind) in [((1,), [5.0 3.0 6.0], [2 3 5]),
+for (tup, rval, rind) in [((1,), [5.0 5.0 6.0], [2 3 5]),
                           ((2,), reshape([6.0,5.0], 2, 1), reshape([5,2], 2, 1)),
                           ((1,2), fill(6.0,1,1),fill(5,1,1))]
     @test findmax(A, tup) == (rval, rind)
     @test findmax!(similar(rval), similar(rind), A) == (rval, rind)
+    @test string(maximum(A, tup)) == string(rval)
```

Comment:
The comparisons using `==` do not work, when the result collections contain `NaN`, because `NaN != NaN`. Comparison with `===` does not compare array elements. 
And I wanted to be able to identify the failed tests by the test values. With a specialized compare function that was not the case ( `@test eq1(maximum(A, tup), rval)` prints its line literally).